dev/io.openliberty.microprofile.reactive.messaging.internal_fat/fat/src/io/openliberty/microprofile/reactive/messaging/fat/kafka/metrics/MetricsTest.java

```diff
@@ -0,0 +1,153 @@
+                        .include(ConnectorProperties.simpleIncomingChannel(KafkaTests.connectionProperties(), ConnectorProperties.DEFAULT_CONNECTOR_ID,
+                                                                           MetricsReceptionBean.CHANNEL_IN, APP_NAME, MetricsTestServlet.EMITTER_TOPIC))
+                        .include(ConnectorProperties.simpleOutgoingChannel(KafkaTests.connectionProperties(), ConnectorProperties.DEFAULT_CONNECTOR_ID,
+                                                                           MetricsDeliveryBean.METRICS_OUTGOING, MetricsTestServlet.OUTGOING_TOPIC))
+                        .include(ConnectorProperties.simpleIncomingChannel(KafkaTests.connectionProperties(), ConnectorProperties.DEFAULT_CONNECTOR_ID,
+                                                                           MetricsReceptionBean.CHANNEL_MESSAGE_IN, APP_NAME, MetricsTestServlet.OUTGOING_TOPIC));
+
+        WebArchive war = ShrinkWrap.create(WebArchive.class, APP_NAME + ".war")
+                        .addAsLibraries(kafkaClientLibs())
+                        .addAsManifestResource(kafkaPermissions(), "permissions.xml")
+                        .addPackage(MetricsTestServlet.class.getPackage())
+                        .addPackage(KafkaTestConstants.class.getPackage())
+                        .addPackage(AbstractKafkaTestServlet.class.getPackage())
+                        .addAsResource(appConfig, "META-INF/microprofile-config.properties");
+
+        ShrinkHelper.exportDropinAppToServer(server, war, SERVER_ONLY);
+        server.startServer();
+    }
+
+    @Test
+    public void testDeliverPayload() throws Exception {
+        // Test that metrics are updated when a payload is delivered via an emitter
+        runTest(server, APP_NAME + "/MetricsTest", "emitterDeliverPayload");
+        HashMap<String, Integer> metrics = getMetrics();
+
+        assertThat("5 messages not detected on channel", metrics.get("metrics-incoming"), equalTo(5));
+//      Assertion disabled as emitters currently don't report metrics per spec
+//        assertTrue("Outgoing and incoming channel counts are not the same",
+//                   metrics.get("mp_messaging_message_count_total{channel=\"MetricsEmitter\",mp_scope=\"base\",}") == metrics
+//                                   .get("mp_messaging_message_count_total{channel=\"metrics-incoming\",mp_scope=\"base\",}"));
+    }
+
+    @Test
+    public void testDeliverMessage() throws Exception {
+        // Test that metrics are updated when a message is delivered via delivery and reception beans
+        runTest(server, APP_NAME + "/MetricsTest", "deliverMessage");
+        HashMap<String, Integer> metrics = getMetrics();
+
+        assertThat("5 messages not detected on channel", metrics.get("metrics-outgoing"), equalTo(5));
+        assertThat("Outgoing and incoming channel counts are not the same", metrics.get("metrics-outgoing"), equalTo(metrics.get("metrics-message-incoming")));
```

Comment:
I didn't notice this until you removed the unused injected `MetricsReceptionBean`, but you're checking that the metrics for the incoming channel have been updated before you confirm that the messages have been received. There's no wait here either, so it would be possible for this check to run before the incoming messages have been processed which would fail the test.

The easiest way to fix this is to put the MetricsReceptionBean back in and use `assertReceivedMessages` to wait for the messages to be received.

You also probably want a separate reception bean for each incoming channel since that will make it easier to assert the right number of messages are received.